qa/build-standalone-example-plugins/build.gradle

```diff
@@ -0,0 +1,58 @@
+import org.elasticsearch.gradle.VersionProperties
+
+import java.util.regex.Matcher
+
+apply plugin: 'elasticsearch.build'
+
+dependencies {
+    testCompile 'commons-io:commons-io:2.6'
+    testCompile gradleTestKit()
+    testCompile (project.project(':test:framework')) {
+        exclude group: "commons-logging"
+    }
+}
+
+task integTest(type: Test) {
```

Comment:
The main issue with moving these tests to their own project outside of `buildSrc` is now we are not tracking the build logic as an input in any way. So we'll continue to use cached results pretty much indefinitely until the sample projects themselves change. Alos, these tests are going to resolve the `build-tools` JAR "externally" now. So a breaking change in `buildSrc` will only manifest when the new JAR is pulled down, not on the build that included the breaking change.

Due to the issues above and the complexity involved in fixing them I think I'm inclined to leave this stuff part of `buildSrc` for simplicity. Essentially, these tests are to ensure changes in `buildSrc` have not caused a regression. For that reason, I think that is where the tests should live, not in a separate project.

Reply:
That's only part of the picture. Changes in build-tools is one thing, but more often some dependency is added, e.x. a new library but publishing for it is not configured for it. 
This was meant for a way to also test that we are publishing everything we need to, and eventually change the release manager so  that we can generate it's configuration from Gradle instead of having a hard-coded list so that we could generate a configuration to publish everything  that has publishing configured. 

One way to solve this is to add the local repo as input, but that would void the cache most of the time, but It might be the only correct way, as  some project might change and have it's publication removed, so the artifact will no longer be there.

Reply:
> but more often some dependency is added, e.x. a new library but publishing for it is not configured for it.

Under what scenario would this happen? The `buildSrc` project cannot depend on any other project, since it's needed at build time. So it's only dependencies would be other third-party libraries that will be available in public repos. If we want better isolation here we should do as you recommend and remove the example projects from the root project completely so that they _have_ to get everything as a binary dependency.

> One way to solve this is to add the local repo as input, but that would void the cache most of the time, but It might be the only correct way

We can just as easily just add `:build-tools` to the test classpath here. This is just as correct.

That said, I still don't quite see the benefits of this being a separate project. It doesn't buy us any additional coverage, but creates additional complexity. What we are testing is `build-tools` so I think these tests should really live there. None of the issues we had with example project testing was caused by the tests being part of the `build-tools` project.

IMO we should defer any significant refactoring here until we tackle the larger issues of third-party plugin development being basically undocumented, our `buildSrc` being littered with loads of internal stuff never intended to be used outside of the elasticsearch project, and our funky quasi-compatibility with Java 8.

Reply:
By "significant refactoring" I mean just the addition of the separate project for example build testing.

Reply:
We agreed with @rjernst a long time ago that we would move these to their own project, the main motivation is that it created a lot of confusion when these broke due to unpublished dependencies. A failure in `build-tools` didn't make it obvious at all what was going on. 

Note that the integ tests are being ran as part of `build-tools` not `buildSrc`. 

That being said I agree that we need to do all those things, but we should not hold up running these tests until all of that happens.

Reply:
> We agreed with @rjernst a long time ago that we would move these to their own project, the main motivation is that it created a lot of confusion when these broke due to unpublished dependencies.

By "own project" I just meant a separate CI build. The concern at the time was performance.

In any case, I still contend that there's no need to move these to a separate project to test dependency isolation. Simply using a `GradleRunner` and not passing `withPluginClasspath()` would have the same affect w/o the need for another project.

If we want to keep the separate QA project that's fine, we just need to track `build-tools` as an input. Let's not try to factor in all the other dependencies the build might use. We weren't testing that before so that hasn't changed. We are only doing assembly here so we don't actually care about the runtime behavior of those JARs, only that they exist.